### PR TITLE
brontide+lnwire+wtwire+zpay32: migrate fuzz tests

### PR DESCRIPTION
**Pull request by morehouse**
This is https://github.com/lightningnetwork/lnd/pull/6784 rebased, with all outstanding comments addressed and linter issues fixed.

Fixes https://github.com/lightningnetwork/lnd/issues/6679

### CODE REVIEW

File: docs/release-notes/release-notes-0.16.0.md [hunk 1]
```diff
@@ -222,6 +225,7 @@ to refactor the itest for code health and maintenance.
 # Contributors (Alphabetical Order)
 
 * Carla Kirk-Cohen
+* Conner Babinchak
```
Comment (Roasbeef) [inline docs/release-notes/release-notes-0.16.0.md:228]: I think you can go ahead and add yourself as will given you contributed some of the fixups.

**Review by Roasbeef — APPROVED** [review]
LGTM 🐅

**Comment by Roasbeef** [comment]
Thx for picking this up! I think we want to hold off on merging this until https://github.com/lightningnetwork/lnd/pull/7126 is in.

File: docs/release-notes/release-notes-0.16.0.md (hunk 1 above)
Comment (morehouse) [reply docs/release-notes/release-notes-0.16.0.md:228]: I'm actually already in this list.

**Review by Crypt-iQ — APPROVED** [review]
if you feel up to it, you could add a fuzzer for `warning` - dont think it should go in this pr since we'd have to re-review it

**Comment by morehouse** [comment]
Rebased.  Should be ready to merge now.

**Pull request merged**